components/board/genericlinux/gpio.go

```diff
@@ -232,5 +260,77 @@ func gpioInitialize(cancelCtx context.Context, gpioMappings map[int]GPIOBoardMap
 			logger:     logger,
 		}
 	}
-	return pins
+	return pins, interrupts, nil
+}
+
+type digitalInterrupt struct {
```

Comment:
[q] would it be better for this to actually be a wrapper struct around `LineWithEvent` and have it satisfy our `DigitalInterrupt` interface? I feel like then `DigitalInterruptByName` could just return this struct and it would resolve the naming concerns you've raised earlier

Reply:
discussed offline, due to the existence of formulas this refactor is more trouble than the benefit it would currently provide. Agreed we should leave as is.